compiler/rustc_codegen_llvm/src/mono_item.rs

```diff
@@ -21,7 +23,14 @@ impl<'tcx> PreDefineMethods<'tcx> for CodegenCx<'_, 'tcx> {
         symbol_name: &str,
     ) {
         let instance = Instance::mono(self.tcx, def_id);
-        let ty = instance.ty(self.tcx, ty::ParamEnv::reveal_all());
+        let DefKind::Static { nested, .. } = self.tcx.def_kind(def_id) else { bug!() };
+        // Nested statics do not have a type, so pick a random type and let `define_static` figure out
```

Comment:
ditto

Reply:
Now I wonder if it's worth factoring out this repeated code.

Reply:
I tried that. It's annoying, and all of this code could have an overhaul in general